### DISCUSSION TITLE
Welcome to ACE Discussions!

### DISCUSSION THREAD

**Discussion by tehKaiN** [AmigaPorts/ACE#132] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Ozzyboshi** [comment]
Hi I didnt know about this github feature.
I am Ozzyboshi, a ACE user , love ACE and waiting for building istructions with Bartman.

**Comment by rveilleux** [comment]
Hi @tehKaiN !
Could you tell us what is your setup for using ACE (which OS, compiler, tools, etc.)? 
I didn't manage to make ACE compile on my Windows 10 machine due to incorrect line endings, and Cygwin is a pain. 

Also bonus question: Do you know about this from Abyss: https://github.com/BartmanAbyss/vscode-amiga-debug
Would it be possible to compile ACE using Visual Studio Code and this toolchain? I tried but it failed to compile the REGARG macro due to asm("d0") not being recognized, and stdio.h being missing, etc. In summary, I failed.

**Comment by tehKaiN** [comment]
Hi,

Yeah, I do use vscode and I know Bartman's excellent extension - I occasionally bug him about some missing features. ;) Recently, I've done a tutorial on how to setup Bartman's toolchain with ACE - it's not yet on master branch, so you need to look at ptplayer branch:

https://github.com/AmigaPorts/ACE/blob/ptplayer_c/docs/installing/compiler.md

This branch adds currently in-development C rewrite of Phx's ptplayer, but some other things have crept into it - including the Bartman's toolchain support. I want to merge in this branch already, but there are still some bugs in .mod playback - otherwise it's fine and you should be able to do everything else without any problems.

Let me know if the instructions are not clear enough for you - having it already set up properly on my machine didn't helped me when writing this guide. ;) ACE is very CMake-oriented, so this is the main supported way, but if you prefer other build systems (e.g. Bartman's makefile) let me know if you need detailed help. If you're not using CMake and toolchain file as specified in my guide, look in `ace/types.h` - there's a `BARTMAN_GCC` macro which you should set when building and `__CODE_CHECKER__` in .vscode/c_cpp_properties.json which you can set to suppress some intellisense errors.

**Reply by rveilleux** [reply]
This is great, thanks a lot @tehKaiN ! By following your instructions + fiddling a bit, I've managed to compile the ACE showcase (although it seems to crash on boot, I haven't managed to run it correctly yet).
I'll keep at it :)
Thanks again!

**Comment by stama182** [comment]
Hi!
Ive been trying to follow the instructions but the only thing I managed to produce is a waitmouse.c.obj file.
How do I test that file in uae?
I use the ninja compiler.

I attached the file so maybe someone can see what Im doing wrong. Ive tried both the lemonamiga and amiga.org forum.
[GoodOldWorld (3).zip](https://github.com/AmigaPorts/ACE/files/14922892/GoodOldWorld.3.zip)

Might I add that I actually work as a programmer, but aint really used to setting up this old pipeline :)
I also made a game for C64 called "eggerland mystery", but then all you needed to do was to open notepad.

**Reply by tehKaiN** [reply]
Hi, welcome to ACE!

I think that semi-worked. Your cmakelists looks as follows:

```cmake
cmake_minimum_required(VERSION 3.12)
project("AllOldWorld")
add_executable(myexec waitmouse.c)
```

so that's trying to create `myexec` executable using waitmouse.c. And if you look at the `build` folder there definitely is `myexec` file, but if you open it with text or hex editor, you'll see the ELF header, so it's still not an Amiga executable. You have to do an extra step to convert it into Amiga hunk file using `elf2hunk` tool, preferably with cmake, as described in [Hello World](https://github.com/AmigaPorts/ACE/blob/master/docs/programming/hello_world.md) section of the docs.

I know it's a bit of reading to actually get something usable, but I strongly encourage you to at least skim through [ACE in a nutshell](https://github.com/AmigaPorts/ACE/blob/master/docs/programming/ace_in_a_nutshell.md) since it describes some initial concepts which are used in "Hello World" docs.

Also, I'd like to point out that ACE docs are JIT-generated, so if you stumble upon something unclear, or undocumented, ask away, and my responses to you will be used to build missing docs. ;)

BTW other websites: I'm not really on lemonamiga and amiga.org, so I completely missed your posts. You can either find me here or on AmigaDev / Amiga Game Dev discord servers.

**Reply by stama182** [reply]
Thank you that did help! 

I didnt know there was a  Hello World section in the docs. I just took them from the beginning and got stuck.

I now have a file called hello.exe, and if I try to launch it in windows it says its for another computer :). I still dont now if it worked, because I havnt been able to test it. If you have a tutorial for that as well Im all ears. I know how to run adf in uae, but this is an alone exe. I tried putting it on a disk with "adf opus" tool, but when I opened the disk in workbench, it appeard empty....

**Reply by tehKaiN** [reply]
Well, you can't run that .exe on Windows, since it's Amiga executable. ;)

Have you followed steps in [Compiler Setup: Setting up the debugger](https://github.com/AmigaPorts/ACE/blob/master/docs/installing/compiler.md#setting-up-the-debugger)? If yes, then you can run your executable by pressing F5 or the run icon from vscode's debugging panel.

Now I see that the docs are a bit chaotic, so I'll add a bit more info after build part in Hello World article. :)
